From ccee5db79d6037c4d752b5f29dc30fb0dfd039bf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timoth=C3=A9e=20Couble?= <timothee.couble@kitware.com>
Date: Fri, 10 Nov 2023 10:03:07 +0100
Subject: [PATCH 1/2] IOPDAL: Always links with pdalcpp

pdal_base and pdal_utils libraries are removed in PDAL 2.6.0
link with pdalcpp instead starting from this version.
---
 IO/PDAL/CMakeLists.txt | 18 ++++++++++++------
 1 file changed, 12 insertions(+), 6 deletions(-)

diff --git a/IO/PDAL/CMakeLists.txt b/IO/PDAL/CMakeLists.txt
index c31af0410f4..b83a9334109 100644
--- a/IO/PDAL/CMakeLists.txt
+++ b/IO/PDAL/CMakeLists.txt
@@ -3,14 +3,20 @@ vtk_module_find_package(PACKAGE PDAL VERSION 2.1)
 set(classes
   vtkPDALReader)
 
-set(pdal_libraries pdal_util)
-if (APPLE OR WIN32)
+set(pdal_libraries)
+if (PDAL_VERSION VERSION_LESS "2.6")
+  set(pdal_libraries pdal_util)
+  if (APPLE OR WIN32)
+    list(APPEND pdal_libraries
+      pdalcpp)
+  else()
+    list(APPEND pdal_libraries
+      pdal_base)
+  endif()
+else ()
   list(APPEND pdal_libraries
     pdalcpp)
-else()
-  list(APPEND pdal_libraries
-    pdal_base)
-endif()
+endif ()
 
 vtk_module_add_module(VTK::IOPDAL
   CLASSES ${classes})
-- 
GitLab


